### ISSUE TITLE
Compatibility with 1.1

### ISSUE THREAD

**Issue by patrick-nicodemus** [owlbarn/owl-plplot#1]
Is there any reason this should be incompatible with owl 1.1? 

If it is compatible, the official opam file should be updated.

There are currently no tests.

**Comment by patrick-nicodemus** [comment]
Ok. I compiled it locally and tested it with some of the examples from the tutorial page and it seems to work.

**Comment by patrick-nicodemus** [comment]
This issue is not closed. I said I tested it. The issue is to update the official opam repository here is up to date.
https://github.com/ocaml/opam-repository/tree/master/packages/owl-plplot

I think the best solution is to make an official new release of owl-plplot (on this git repo) and then ask the opam maintainers to accept an updated opam file in their repository.

**Comment by patrick-nicodemus** [comment]
@jzstark 
Can you please reopen the issue or if possible give me permissions for this repository?

**Comment by patrick-nicodemus** [comment]
I am not an experienced ocaml developer but this page seems out of date.
https://opam.ocaml.org/packages/owl-plplot/

The source URL for the latest version is not appropriately decoupled/redirected to this current repository, note that it says https://github.com/owlbarn/owl/releases/download/1.0.2/owl-1.0.2.tbz 
The owl version required is either 1.0.2 or 1.0.2-1.

My understanding is that when a user writes "opam install owl-plplot", this file controls the installation.

https://github.com/ocaml/opam-repository/blob/a799daa56b1b3824d067c7cc92be627eef88aafe/packages/owl-plplot/owl-plplot.1.0.2/opam

**Comment by mseri** [comment]
@patrick-nicodemus you are right. The best idea would be to make a release depending on owl >= 1.1 and pointing to a 1.1 release in this repo

**Comment by Weissstreifen** [comment]
Hello, did something happen in the meantime? I was trying to use the plotting library but the dependency locks me out of using ocaml 5.2.0.

Thanks for your hard work!

**Comment by patrick-nicodemus** [comment]
I don't think there are updates so far, I think someone with permissions has to do it. If you clone the repo and change the minimum version number in the opam file you can install it locally.

Plplot itself is well documented and there is OCaml sample code available. It is a very imperative style, not super idiomatic, but super readable. The functions in owl plplot file are more convenience functions so you don't have to write your own histogram routine for plplot, not a fundamental coupling necessary to use plplot, you can use plplot directly without much hassle. I was pleasantly surprised at the elegant design of the library compared to Matplotlib.

**Comment by mseri** [comment]
I think the release has been made already enabling ocaml < 5.2. The problem is that owl itself is not compatible with ocaml 5.2 and it is nontrivial to make it work: https://github.com/owlbarn/owl/issues/657

As @patrick-nicodemus mentions, if you don’t need owl you can still use the ocaml `plplot` package: https://ocaml.org/p/plplot/5.12.0

**Comment by dayfine** [comment]
Had to do:
```sh
sh opam repository set-url default https://github.com/ocaml/opam-repository.git && opam update
```
